### PR TITLE
Show employees List with bloc using dummy data

### PR DESCRIPTION
**Pull request by HKayed**

### CODE REVIEW

**Review by TugbaDalmaz — APPROVED** [review]

File: basic_flutter/lib/repository/data_provider.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        dynamic employeeListJson = json.decode(resp.body);
+        for (dynamic currentEmployee in employeeListJson) {
+          response.add(Employee.fromJson(currentEmployee));
+        }
+      }
+    });
+    return response;
+  }
+
+  static void postEmployeeData(String url) async {
```
Comment (nicotheod) [inline basic_flutter/lib/repository/data_provider.dart:21]: I assume this is in progress?

File: basic_flutter/lib/repository/data_provider.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+import '../data/employee.dart';
+
+class DataProvider {
+  late List<Employee> response;
+
+  Future getEmployeesList(String endpoint) async {
```
Comment (nicotheod) [inline basic_flutter/lib/repository/data_provider.dart:8]: define return type

File: basic_flutter/lib/repository/data_provider.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    await http.post(Uri.parse(endpoint),
+        body: [], headers: {"Content-Type": "application/json"}).then((resp) {
```
Comment (nicotheod) [inline basic_flutter/lib/repository/data_provider.dart:10]: this is an async function you don't need .then here, but you can do it like that, if you want.

**Review by nicotheod — CHANGES_REQUESTED** [review]
There are some parts on the wrong place and some smaller errors are repeating itself. I marked them once but please look on the hole code for it :) I hope it will help you!

File: basic_flutter/lib/repository/data_provider.dart (hunk 3 above)
Comment (HKayed) [reply basic_flutter/lib/repository/data_provider.dart:10]: I still see it a better fit to use this future execution. I'm saving initializing a new variable which I'm already getting from the response of the request.

File: basic_flutter/lib/business_logic/cubits/response_cubit.dart [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      : super(ResponseState(DataLoadingStates.dataLoading, dataProvider)) {
+    getStateData();
+  }
+
+  void getStateData() async {
+    try {
+      final employeeList =
+          await dataProvider.getEmployeesList(EmployeesListScreen.url);
+      if (employeeList.isNotEmpty) {
+        emit(ResponseState(DataLoadingStates.dataLoaded, dataProvider));
```
Comment (nicotheod) [inline basic_flutter/lib/business_logic/cubits/response_cubit.dart:20]: No need to emit the DataProvider. You shouldn't use it in the UI part.

File: basic_flutter/lib/presentation/screens/employees_list_screen.dart [hunk 5]
```diff
@@ -0,0 +1,82 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter/material.dart';
+
+import '/repository/data_provider.dart';
+import '../widgets/list_item.dart';
+import '/business_logic/cubits/response_cubit.dart';
+import '/business_logic/cubits/response_state.dart';
+import '/data/routes.dart';
+
+class EmployeesListScreen extends StatefulWidget {
```
Comment (nicotheod) [inline basic_flutter/lib/presentation/screens/employees_list_screen.dart:10]: Why stateful?

**Review by nicotheod — CHANGES_REQUESTED** [review]
Some old comments are also up to date

File: basic_flutter/lib/presentation/screens/employees_list_screen.dart (hunk 5 above)
Comment (HKayed) [reply basic_flutter/lib/presentation/screens/employees_list_screen.dart:10]: @nicotheod  I need the stateful widget for the actions in the appbar. It surely didn't make sense at that time. Should I add a comment for this?

**Pull request merged**